File: day_06/day6_2.py
import sys

def isMarker(text):
    return len(text) == len(set(text))
    
def parseBuffer(buffer):
    for i in range(len(buffer) -14):
        marker = isMarker(buffer[i:i+14])
        if marker:
            return i+14 

for line in sys.stdin:
    print(parseBuffer(line))

File: day_06/day6_2.js
process.stdin.on('data', data => {
    console.log(data.toString().split('\r\n').filter(o=>o).map(parseBuffer));
    process.exit();
  });

function isMarker(text){
    return text.length == new Set(text).size
}

function parseBuffer(buffer){
    for( let i=0; i < buffer.length-14; i++){
        if(isMarker(buffer.slice(i,i+14))) return i + 14   
    }
}

// cat .\input.txt | node .\day6_2.js 